std/algorithm/iteration.d

```diff
@@ -885,15 +893,18 @@ template each(alias pred = "a")
         (!isForwardRange!R || isDynamicArray!R) &&
         (isForeachUnaryIterable!R || isForeachBinaryIterable!R);
 
-    void each(Range)(Range r)
+    Flag!"each" each(Range)(Range r)
     if (isRangeIterable!Range && !isForeachIterable!Range)
     {
         debug(each) pragma(msg, "Using while for ", Range.stringof);
         static if (isRangeUnaryIterable!Range)
         {
             while (!r.empty)
             {
-                cast(void) unaryFun!pred(r.front);
+                static if (!is(typeof(unaryFun!fun(r.front)) == Flag!"each"))
```

Comment:
The `!` is a cognitive problem. Negations in general should be avoided. I missed it myself the first time through, and got mildly confused for a moment.